avm/ptn/deployment-script/import-image-to-acr/main.bicep

```diff
@@ -0,0 +1,255 @@
+
+resource avmTelemetry 'Microsoft.Resources/deployments@2024-03-01' = if (enableTelemetry) {
+  name: '46d3xbcp.ptn.deploymentscript-importimagetoacr.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+resource acr 'Microsoft.ContainerRegistry/registries@2023-07-01' existing = {
+  name: acrName
+}
+
+resource newManagedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' = if (!useExistingManagedIdentity) {
+  name: managedIdentityName
+  location: location
+  tags: tags
+}
+
+resource existingManagedIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' existing = if (useExistingManagedIdentity) {
+  name: managedIdentityName
+  scope: resourceGroup(existingManagedIdentitySubId, existingManagedIdentityResourceGroupName)
+}
+
+resource rbac 'Microsoft.Authorization/roleAssignments@2020-08-01-preview' = if (!empty(rbacRoleNeeded)) {
+  name: guid(acr.id, rbacRoleNeeded, useExistingManagedIdentity ? existingManagedIdentity.id : newManagedIdentity.id)
+  scope: acr
+  properties: {
+    roleDefinitionId: resourceId('Microsoft.Authorization/roleDefinitions', rbacRoleNeeded)
+    principalId: useExistingManagedIdentity
+      ? existingManagedIdentity.properties.principalId
+      : newManagedIdentity.properties.principalId
+    principalType: 'ServicePrincipal'
+  }
+}
+
+resource importImage 'Microsoft.Resources/deploymentScripts@2023-08-01' = {
+  name: 'ACR-Import-${name}-${last(split(replace(image,':','-'),'/'))}'
+  location: location
+  tags: tags
+  identity: {
+    type: 'UserAssigned'
+    userAssignedIdentities: {
+      '${useExistingManagedIdentity ? existingManagedIdentity.id : newManagedIdentity.id}': {}
+    }
+  }
+  kind: 'AzureCLI'
+  dependsOn: [rbac]
+  properties: union(
+    {
+      forceUpdateTag: forceUpdateTag
+      azCliVersion: '2.59.0'
+      timeout: 'PT30M' // set timeout to 30m
+      retentionInterval: 'PT1H' // cleanup after 1h
+      environmentVariables: [
+        {
+          name: 'acrName'
+          value: acrName
+        }
+        {
+          name: 'imageName'
+          value: image
+        }
+        {
+          name: 'overwriteExistingImage'
+          value: toLower(string(overwriteExistingImage))
+        }
+        {
+          name: 'initialDelay'
+          value: '${string(initialScriptDelay)}s'
+        }
+        {
+          name: 'retryMax'
+          value: string(retryMax)
+        }
+        {
+          name: 'retrySleep'
+          value: '5s'
+        }
+      ]
+      scriptContent: '''#!/bin/bash
```

Comment:
Should be moved into a dedicated bash script that's loaded into the Bicep file. 
The reason for doing this: Maintainability at no extra cost. For reference: https://github.com/Azure/bicep-registry-modules/tree/main/avm/ptn/lz/sub-vending/scripts